helix-core/src/test/java/org/apache/helix/integration/rebalancer/TestClusterInMaintenanceModeWhenReachingMaxPartition.java

```diff
@@ -114,10 +115,10 @@ public void testDisableCluster() throws Exception {
       _participants.get(i).syncStop();
     }
 
-    Thread.sleep(1000L);
-    maintenanceSignal = _dataAccessor.getProperty(_dataAccessor.keyBuilder().maintenance());
-    Assert.assertNotNull(maintenanceSignal);
-    Assert.assertNotNull(maintenanceSignal.getReason());
+    boolean result = TestHelper.verify(() -> {
```

Comment:
Assert it

Reply:
Good catch. This is actually addressed in https://github.com/apache/helix/pull/1447/files. This file change also belongs there